src/NUnitFramework/Directory.Build.props

```diff
@@ -35,6 +39,41 @@
     <EmbedUntrackedSources>true</EmbedUntrackedSources>
   </PropertyGroup>
 
+    <!-- .NET Core 3.1 -->
+    <PropertyGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+      <AssemblyConfiguration>.NET Core 3.1 $(Configuration)</AssemblyConfiguration>
+    </PropertyGroup>
+
+    <!-- .NET Framework 4.6.2 -->
+    <PropertyGroup Condition="'$(TargetFramework)' == 'net462'">
+      <AssemblyConfiguration>.NET Framework 4.6.2 $(Configuration)</AssemblyConfiguration>
+    </PropertyGroup>
+
+    <!-- .NET 5.0 -->
+    <PropertyGroup Condition="'$(TargetFramework)' == 'net5.0'">
+      <AssemblyConfiguration>.NET 5.0 $(Configuration)</AssemblyConfiguration>
+    </PropertyGroup>
+
+    <!-- .NET 6.0 -->
+    <PropertyGroup Condition="'$(TargetFramework)' == 'net6.0'">
+      <AssemblyConfiguration>.NET 6.0 $(Configuration)</AssemblyConfiguration>
+    </PropertyGroup>
+
+    <!-- .NET 7.0 -->
+    <PropertyGroup Condition="'$(TargetFramework)' == 'net7.0'">
+      <AssemblyConfiguration>.NET 7.0 $(Configuration)</AssemblyConfiguration>
+    </PropertyGroup>
+
+    <!-- .NET 8.0 -->
+    <PropertyGroup Condition="'$(TargetFramework)' == 'net8.0'">
+      <AssemblyConfiguration>.NET 8.0 $(Configuration)</AssemblyConfiguration>
+    </PropertyGroup>
+
+    <!-- .NET 9.0 -->
+    <PropertyGroup Condition="'$(TargetFramework)' == 'net9.0'">
+      <AssemblyConfiguration>.NET 9.0 $(Configuration)</AssemblyConfiguration>
+    </PropertyGroup>
+
```

Comment:
Nice forward thinking, but to prevent use not setting this when .net10 comes out.
Can I suggest to use [Choose](https://learn.microsoft.com/en-us/visualstudio/msbuild/choose-element-msbuild?view=vs-2022) instead? This allows an `Otherwise` clause where you can fail the build.

```xml
<Choose>
  <When Condition="'$(TargetFramework)' == 'net462'">
    <PropertyGroup>
      <AssemblyConfiguration>.NET Framework 4.6.2 $(Configuration)</AssemblyConfiguration>
    </PropertyGroup>
  </When>
  <When Condition="'$(TargetFramework)' == 'net8.0'">
    ...
  </When>
  <Otherwise>
    <Error Text="MissingAssemblyConfiguration for $(TargetFramework)" />
  <Otherwise>
</Choose>
```

Alternatively add a conditional error:
```xml
<Error Condition="'$(AssemblyConfiguration)' == ''" Text="MissingAssemblyConfiguration for $(TargetFramework)" />
```

Reply:
I tried to add this, which looked very nice, but ended with a msbuild error when running restore.  VS says there is a syntax  error here somewhere, but I can't see it.  Might be something small.  Have commented out for now, but would be great if you had a look at it @manfred-brands .

Reply:
Unfortunately the `Error` task cannot be placed in a `Choose`.
I added a commit to go with option B.

If a non-supported target is used it fails build with:
```
src\NUnitFramework\Directory.Build.props(71,5): error : Missing AssemblyConfiguration for net6.0-windows7 [src\NUnitFramework\windows-tests\windows-tests.csproj::TargetFramework=net6.0-windows7]
src\NUnitFramework\Directory.Build.props(71,5): error : Missing AssemblyConfiguration for net8.0-windows10.0.19041.0 [src\NUnitFramework\windows-tests\windows-tests.csproj::TargetFramework=net8.0-windows10.0.19041.0]
```

I fixed those by use `StartsWith` in the Conditions.

Reply:
It should be this one:
```
Author: Manfred Brands <manfred-brands@users.noreply.github.com>
Date: 14/05/2024 17:47:34
Commit hash: b6448846040142fa792d9d900ad9e9a63ab0ee5b

Raise Error when compiling for a TargetFramework with no AssemblyConfiguration

.NET 7.0 is out of support as of May 14th 2024.
```

It is part of your merge commit.

Reply:
Thanks!